tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/UserEvents/UserEventsCommon.cs

```diff
@@ -9,25 +9,35 @@
 using System.Runtime.CompilerServices;
 using System.Security.Cryptography;
 using System.Text;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.Telemetry.Metrics;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore.UserEvents;
 
 internal static class UserEventsCommon
 {
-    internal static string? GetId(IIdentityUser? user)
-    {
-        return user?.Id?.ToString() ?? user?.Email ?? user?.UserName;
-    }
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(UserEventsCommon));
 
-    internal static unsafe string? GetAnonId(string id)
-    {
-        using var hash = SHA256.Create();
-        var byteArray = hash.ComputeHash(Encoding.UTF8.GetBytes(id));
+    internal static string? GetId(IIdentityUser? user) => user?.Id?.ToString();
 
+    internal static string? GetLogin(IIdentityUser? user) => user?.UserName ?? user?.Email;
+
+    internal static unsafe string Anonymize(string id)
```

Comment:
Does this really need to be unsafe? 🤔

Reply:
unfortunately `stackalloc` seems to need the `unsafe` keyword..

Reply:
`stackalloc` doesn't need the `unsafe` keyword _as long_ as you explicitly set the variable type to `Span<T>` and don't use `var` 🙂 https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc

Reply:
oh cool I didnt know that thanks
although it's gotta be compatible net461... so doesnt seem like I can really declare a `Span<T>` here